spec.md

```diff
@@ -5,4599 +5,421 @@ keywords: registry, on-prem, images, tags, repository, distribution, api, advanc
-The registry notifies the build server that the upload has already been partially attempted.
-The build server responds by only sending the remaining data to complete the image file.
+### Official Certification
 
-### Resumable Pull
+Registry providers can self-certify by submitting conformance results to [opencontainers/oci-conformance](https://github.com/opencontainers/oci-conformance).
 
-Company X is having more connectivity problems but this time in their deployment datacenter.
-When downloading an image, the connection is interrupted before completion.
-The client keeps the partial data and uses http `Range` requests to avoid downloading repeated data.
+### Workflow Categories
 
-### Layer Upload De-duplication
+#### Pull
 
-Company Y's build system creates two identical layers from build processes A and B.
-Build process A completes uploading the layer before B.
-When process B attempts to upload the layer, the registry indicates that its not necessary because the layer is already known.
+##### Pulling Blobs
 
-If process A and B upload the same layer at the same time, both operations will proceed and the first to complete will be stored in the registry (Note: we MAY modify this to prevent dogpile with some locking mechanism).
+To pull a blob, perform a `GET` request to a url in the following form:
+[2a](#Endpoints) `/v2/<name>/blobs/<digest>`
 
-## Changes
+`<name>` is the namespace of the repository, and `<digest>` is the blob's digest.
 
-The V2 specification has been written to work as a living document, specifying only what is certain and leaving what is not specified open or to future changes.
-Only non-conflicting additions SHOULD be made to the API and accepted changes SHOULD avoid preventing future changes from happening.
+A GET request to an existing blob URL MUST provide the expected blob, with a reponse code that MUST be `200 OK`.
 
-The [changes.md](changes.md) doc SHOULD be updated when changes are made to the specification, indicating what is different.
-Optionally, we MAY start marking parts of the specification to correspond with the versions enumerated here.
+If the blob is not found in the registry, the response code MUST be `404 Not Found`.
 
-## Overview
+##### Pulling manifests
 
-This section covers client flows and details of the API endpoints.
-The URI layout of the new API is structured to support a rich authentication and authorization model by leveraging namespaces.
-All endpoints will be prefixed by the API version and the repository name:
+To pull a manifest, perform a `GET` request to a url in the following form:
+[3a](#Endpoints) `/v2/<name>/manifests/<reference>`
 
-```HTTP
-/v2/<name>/
-```
+`<name>` refers to the namespace of the repository. `<reference>` MUST be either (a) the digest of the manifest or (b) a tag name.
 
-For example, an API endpoint that will work with the `library/ubuntu` repository, the URI prefix will be:
+The `<reference>` MUST NOT be in any other format.
```

Comment:
Seems like the two sentences referring to 'reference' should be in the same paragraph?

Reply:
:+1: